test/ggsn_proxy_SUITE.erl

```diff
@@ -859,6 +867,45 @@ create_pdp_context_request_accept_new(Config) ->
     meck_validate(Config),
     ok.
 
+%%--------------------------------------------------------------------
+path_maint() ->
+    [{doc, "Check that Echo requests are sent"}].
+path_maint(Config) ->
+    GSNs = lists:foldl(
+	     fun(X, M) -> maps:put(proplists:get_value(X, Config), X, M) end,
+	     #{}, [client_ip, test_gsn, proxy_gsn, final_gsn]),
+
+    {GtpC0, _, _} = create_pdp_context(random, Config),
+    ct:sleep(500),
+
+    {GtpC1, _, _} = create_pdp_context(random, GtpC0),
+    ct:sleep(500),
+
+    {GtpC2, _, _} = create_pdp_context(random, GtpC1),
+    ct:sleep(500),
+
+    Pings = lists:foldl(
+	      fun({_, {ergw_gtp_c_socket, send_request, [_, IP, _, _, _, #gtp{type = echo_request}, _]}, _}, M) ->
+		      Key = maps:get(IP, GSNs),
+		      maps:update_with(Key, fun(Cnt) -> Cnt + 1 end, 1, M);
+		 (_, M) ->
+		      M
+	      end, #{}, meck:history(ergw_gtp_c_socket)),
+
+    delete_pdp_context(GtpC0),
+    delete_pdp_context(GtpC1),
+    delete_pdp_context(GtpC2),
+
+    ok = meck:wait(?HUT, terminate, '_', ?TIMEOUT),
+    wait4tunnels(?TIMEOUT),
+
+    meck_validate(Config),
+
+    ?equal(3, map_size(Pings)),
```

Comment:
Why did proxy send Echo request to 3 different IP? I thought it would only send to GGSN?

Reply:
The config affects all path that are managed by ergw. And that is PGW->proxy, proxy->client, and proxy->PGW.
The client->proxy path is managed by the test client and that does not have active path maintenance.